tests/lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,422 @@
+/**
+ * @fileoverview Prefers object spread property over Object.assign
+ * @author Sharmila Jesupaul
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+const rule = require("../../../lib/rules/prefer-object-spread");
+
+const RuleTester = require("../../../lib/testers/rule-tester");
+
+const parserOptions = {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+        experimentalObjectRestSpread: true
+    }
+};
+
+const ruleTester = new RuleTester({ parserOptions });
+
+ruleTester.run("prefer-object-spread", rule, {
+    valid: [
+        "Object.assign()",
+        "let a = Object.assign(a, b)",
+        "Object.assign(a, b)",
+        "let a = Object.assign(b, { c: 1 })",
+        "let a = Object.assign({}, ...b)",
```

Comment:
Should this be valid? Wondering if the correct code should be `let a = Object.assign(...b);`, or am I missing something? (Compare with the test case 2 lines below this one)

Reply:
Statement and expression positions differ in terms of how the autofixer outputs parens, but good call, this one should be a spread.